test/ipns-pubsub.js

```diff
@@ -107,64 +108,43 @@ const subscribeToReceiveByPubsub = (nodeA, nodeB, idA, idB, callback) => {
   const topic = `${namespace}${base64url.encode(keys.routingKey.toBuffer())}`
 
   // try to resolve a unpublished record (will subscribe it)
-  nodeB.api.name.resolve(idA, (err) => {
+  try {
+    await nodeB.api.name.resolve(idA)
```

Comment:
Instead of try/catch:

```suggestion
    await expect(nodeB.api.name.resolve(idA)).to.be.rejected()
```

The other assertions can then continue as normal without being in a `catch` block that may not get executed as @vasco-santos points out.

Reply:
Cool stuff. I didn't know about that!